axelor-production/src/main/java/com/axelor/apps/production/service/operationorder/OperationOrderServiceImpl.java

```diff
@@ -439,4 +439,47 @@ public List<OperationOrder> getReversedSortedOperationOrderList(
 
     return Lists.reverse(getSortedOperationOrderList(operationOrders));
   }
+
+  @Override
+  public boolean isTimeBeforeNextOperation(OperationOrder operationOrder) {
+    boolean isTimeBeforeNextOperation = false;
+    ManufOrder manufOrder = operationOrder.getManufOrder();
+
+    if (operationOrder.getStatusSelect() == OperationOrderRepository.STATUS_PLANNED
+        && operationOrder.getPriority() == manufOrder.getOperationOrderMaxPriority()) {
+
+      OperationOrder previousPriorityOperationOrder = getPreviousPriorityOperationOrder(manufOrder);
+      if (previousPriorityOperationOrder != null
+          && previousPriorityOperationOrder.getProdProcessLine() != null) {
+        isTimeBeforeNextOperation = checkTimeBeforeNextOperation(previousPriorityOperationOrder);
+      }
+    }
+    return isTimeBeforeNextOperation;
+  }
+
+  protected OperationOrder getPreviousPriorityOperationOrder(ManufOrder manufOrder) {
+    return manufOrder.getOperationOrderList().stream()
+        .filter(
+            op ->
+                (op.getPriority() < manufOrder.getOperationOrderMaxPriority())
+                    && op.getStatusSelect() == OperationOrderRepository.STATUS_FINISHED)
+        .max(getOperationOrderComparator())
+        .orElse(null);
+  }
+
+  protected Comparator<OperationOrder> getOperationOrderComparator() {
+    return Comparator.comparing(OperationOrder::getPriority)
+        .thenComparing(OperationOrder::getRealEndDateT, (d1, d2) -> d1.compareTo(d2));
```

Comment:
I don't think we need `(d1, d2) -> d1.compareTo(d2)` explicit comparator if we are passing same key extractor